### DISCUSSION TITLE
How to set "block_1","block_2",...,and "nodelist_1","nodelist_2"  in Gmsh

### DISCUSSION THREAD

**Discussion by IntePNum** [peridigm/peridigm#338] [General]
Dear professors and doctors,

I want to use the Gmsh to generate the input file of Peridigm. 
I want to known how to set "block_1","block_2",...,and "nodelist_1","nodelist_2"  in Gmsh?
![1](https://github.com/peridigm/peridigm/assets/108223279/195df846-fa72-4137-8965-6112d8c39189)

**Comment by oldninja** [comment]
I think you would need to read the gmsh documentation for this. Keep in mind that elements in FE are the equivalent to the peridigm material points.

**Reply by IntePNum** [reply]
Ok, I know nothing about gmsh, i will read it

**Reply by oldninja** [reply]
You will need to create physical entities, if I remember correctly, then create sets for the elements.

**Reply by IntePNum** [reply]
Thank you for your answer. I think i should read the user guidline of gmsh,hhhhh!

**Comment by oldninja** [comment]
Another note, for such a simple geometry, it may be easier to just create a python script and do a text file instead. Unless you are trying to do something more complex on the inside of the cubes. Also, try to keep uniform spacing between discrete points so that you get a minimum of 3dx for you horizon.

**Reply by IntePNum** [reply]
It is a wonderful advice. But how to keep uniform spacing between points. It is amazing

**Reply by oldninja** [reply]
See this
 https://gitlab.onelab.info/gmsh/gmsh/blob/master/tutorials/python/t3.py

**Reply by IntePNum** [reply]
OK, Thank you for the recommend. 
I am new to gmsh and Peridigm.
I will learn them slowly in my future work.
does it has examples in the Peridigm package about their apllication?

**Reply by oldninja** [reply]
There aren't any gmsh examples in Peridigm.

**Reply by oldninja** [reply]
Here is one example of a plate with a hole- I'm sure you can follow the steps and change it for it to fit your problem:
[plate_with_hole.zip](https://github.com/peridigm/peridigm/files/11645888/plate_with_hole.zip)

**Reply by IntePNum** [reply]
you are so kind.
Thanks for your example!
It is very useful for me.
best wishes!!
: ) ^^

**Reply by paulordutra** [reply]
Hey @oldninja, how do you define nodelists in gmsh? Is there any specific command for this?

**Reply by oldninja** [reply]
I generally create physical tabs for the elements.

**Reply by paulordutra** [reply]
OK, I will try this approach. Thanks for the answer.
